lib/team_result_parser.rb

```diff
@@ -1,13 +1,16 @@
 require "csv"
-require './lib/team_result'
+require_relative './team_result'
+require_relative './league_parser'
 class TeamResultParser
   attr_reader :game_data_by_team
   def initialize
     @game_data_by_team = []
+    @league_parser = LeagueParser.new
```

Comment:
brilliant move instantiating @league_parser here to be able to call on the teams.csv data without passing in an argument!